tests/integration/security/rbac/v2/basic/main_test.go

```diff
@@ -64,6 +64,9 @@ func setupConfig(cfg *istio.Config) {
 	}
 	isMtlsEnabled = cfg.IsMtlsEnabled()
 	cfg.Values["sidecarInjectorWebhook.rewriteAppHTTPProbe"] = "true"
+	// Turn on mTLS since RBAC is usually used with mTLS.
+	cfg.Values["global.mtls.enabled"] = "true"
```

Comment:
This ignores the value from `cfg.IsMtlsEnabled()` which doesn't seem right. The value `isMtlsEnabled` is used in some test cases which I suppose we should run those test cases with both mTLS enabled and disabled, that's why we put the variable `isMtlsEnabled` in those tests right? Otherwise the test case should use constant `true` and making mTLS as a required dependency for them.

Reply:
Good catch. I'm debating weather we should turn on mtls by default (and remove `isMtlsEnabled = cfg.IsMtlsEnabled()`) or let the test runner set the global environment. I think currently it just lets each test suite set its own environment (https://storage.googleapis.com/istio-prow/logs/istio-integ-k8s-tests-master/215/build-log.txt). 

@ozevren @nmittler The old integration test runner runs both mtls (https://testgrid.k8s.io/istio-postsubmits-master#circleci-e2e-pilot-auth-v1alpha3-v2) and non-mtls (https://testgrid.k8s.io/istio-postsubmits-master#circleci-e2e-pilot-noauth-v1alpha3-v2) tests. Are we planning to do something like this for the new test framework or does each test suite define its own environment and cannot be overridden by the test runner (`go test...`)?

Reply:
@yangminzhu I have removed enabling mtls in this test suite by default (`cfg.Values["global.mtls.enabled"] = "true"`) for now.

Reply:
This is a bit of an open area at the moment. Right now, the test framework gives you the ultimate flexibility. However, to improve runtime of tests, some folks want to do setup once (externally) and run the tests against that setup. Today, we're accomplishing this via the "nodeploy" flag, which allows skipping of deployment.

I am not sure if we have a good testing plan in place at the moment to run tests against various configurations. IMHO, the right approach would be to:

- Come up with a comprehensive, well known set of deployment configurations (i.e. based on a matrix of various top-level settings, such as mTls v.s non-Mtls or other dimensions). Hopefully, this 
- Make the Istio component have the ability to deploy this closed-set enumeration of deployment configurations. Pick one as the default.
- Label tests, so that they can be filtered out automatically based on the deployment config being used.
- Allow the user to specify which deployment config they'd like to use.
- Keep supporting --nodeploy, so that the deployment can be done once on the outside, for CI systems (or for running all tests together, locally).

This should give us best of both worlds. It allows a user to run "go test" and get a very good amount of coverage, it allows running the same tests to be run against different, supported configurations, and it also allows running tests against an externally deployed environment.

Reply:
@ozevren Running tests in 2 environments (mtls and non-mtls) will be better and test writers do not have to write duplicated code. I don't see any significant drawbacks from that. Why are we not doing this?

Reply:
I don't know what you mean by that. You can certainly do it today with the test framework. In different ways in fact. For example, You can deploy istio within the context of a test. You can also implement your core test logic as a common function that you can call from your test. Then, you can have two tests, one deploying differently, but calling the same method. This is one way. What I am suggesting above is another way.

As for why we're not doing that, is because nobody have taken over the work to come up with the right design and implement it yet.